go/host/enclave/guardian.go

```diff
@@ -515,45 +517,81 @@ func (g *Guardian) periodicBatchProduction() {
 	}
 }
 
-func (g *Guardian) periodicRollupProduction() {
+func (g *Guardian) periodicRollupProduction() { //nolint:gocognit
 	defer g.logger.Info("Stopping rollup production")
 
-	interval := g.rollupInterval
-	if interval == 0 {
-		interval = 3 * time.Second
+	// rollup at least at every g.rollupInterval
+	rollupInterval := g.rollupInterval
+	if rollupInterval == 0 {
+		g.logger.Crit("invalid rollup interval 0")
 	}
-	rollupTicker := time.NewTicker(interval)
-	// attempt to produce rollup every time the timer ticks until we are stopped/interrupted
+	rollupIntervalBasedTicker := time.NewTicker(rollupInterval)
```

Comment:
I think having two timers is hard to reason about, would need to delve into the timer implementation to convince myself it's safe.

Couldn't you have a `lastSuccessfulRollup := time.Now()` that gets set to `Now()` after every successful `PublishRollup()`.

Then you can just have the quick timer, and have it check `time.Since(lastSuccessfulRollup) > rollupInterval` as well as the size check.

Reply:
agree. 
That's easier to reason about.
A single ticker executing every l1 block, with a bit of logic to decide whether to publish a rollup.
The logic is:
- either the timeout since the last publishing was exceeded
- there is enough data